### ISSUE TITLE
Link errors in Windows with latest nightly

### ISSUE THREAD

**Issue by c-mendoza** [openframeworks/openFrameworks#7470]
I'm getting link errors with the latest nightly in Windows 10:

```
2>openframeworksLib.lib(ofDirectShowPlayer.obj) : error LNK2001: unresolved external symbol IID_IBaseFilter
2>openframeworksLib.lib(ofDirectShowPlayer.obj) : error LNK2001: unresolved external symbol IID_IMediaSeeking
2>openframeworksLib.lib(ofDirectShowPlayer.obj) : error LNK2001: unresolved external symbol IID_IGraphBuilder
2>openframeworksLib.lib(ofDirectShowPlayer.obj) : error LNK2001: unresolved external symbol IID_IMediaControl
2>openframeworksLib.lib(ofDirectShowPlayer.obj) : error LNK2001: unresolved external symbol IID_IMediaEvent
2>openframeworksLib.lib(ofDirectShowPlayer.obj) : error LNK2001: unresolved external symbol IID_IMediaPosition
2>openframeworksLib.lib(ofDirectShowPlayer.obj) : error LNK2001: unresolved external symbol IID_IBasicAudio
2>openframeworksLib.lib(ofDirectShowPlayer.obj) : error LNK2001: unresolved external symbol MEDIATYPE_Video
2>openframeworksLib.lib(ofDirectShowPlayer.obj) : error LNK2001: unresolved external symbol MEDIASUBTYPE_RGB24
2>openframeworksLib.lib(ofDirectShowPlayer.obj) : error LNK2001: unresolved external symbol MEDIASUBTYPE_RGB32
2>openframeworksLib.lib(ofDirectShowPlayer.obj) : error LNK2001: unresolved external symbol CLSID_FilterGraph
2>openframeworksLib.lib(ofDirectShowPlayer.obj) : error LNK2001: unresolved external symbol FORMAT_VideoInfo
2>openframeworksLib.lib(ofDirectShowPlayer.obj) : error LNK2001: unresolved external symbol CLSID_SampleGrabber
2>openframeworksLib.lib(ofDirectShowPlayer.obj) : error LNK2001: unresolved external symbol IID_ISampleGrabber
2>openframeworksLib.lib(ofDirectShowPlayer.obj) : error LNK2001: unresolved external symbol CLSID_NullRenderer
2>bin\videoPlayerExample.exe : fatal error LNK1120: 15 unresolved externals
2>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(1096,5): error MSB6006: "link.exe" exited with code 1120.
```
Just in case it helps, I rolled back to f8da29f3cf7463386eb627049e3d82f4cf7809a1 and that has no link errors.

**Comment by ofTheo** [comment]
Here is a diff of the changes between that commit and now: 
https://github.com/openframeworks/openFrameworks/compare/f8da29f3cf7463386eb627049e3d82f4cf7809a1...master#diff-bca6f4d4670aee0557f1ae4e1e7569a6dc52e6a412f071f04d2f5b4dbf9256d9R16

I wonder if it could be this change: 
<img width="1603" alt="image" src="https://user-images.githubusercontent.com/144000/230135201-a4279f6f-04c7-4c62-9120-196d4e2cb9af.png">


Or maybe: 
<img width="1605" alt="image" src="https://user-images.githubusercontent.com/144000/230135672-2ace8a29-7027-481d-b859-57e92c92bd1a.png">

Could you try replacing ofUtils.cpp with: 
https://github.com/openframeworks/openFrameworks/blob/977cbf3797a13a2ec39121405b2a82a35e418de2/libs/openFrameworks/utils/ofUtils.cpp 

And see if that fixes it? 

Or try replacing ofVideoGrabber with these two files:
https://github.com/openframeworks/openFrameworks/blob/11366f3b56589b267ed5abf0ee7ad26ea377e488/libs/openFrameworks/video/ofVideoGrabber.h
https://github.com/openframeworks/openFrameworks/blob/11366f3b56589b267ed5abf0ee7ad26ea377e488/libs/openFrameworks/video/ofVideoGrabber.cpp

**Comment by NickHardeman** [comment]
@c-mendoza I have the same issue using VS2019 and the latest nightly. At the top of ofDirectShowPlayer.cpp, can you add 
`#pragma comment(lib, "Strmiids.lib")`

**Comment by dimitre** [comment]
I'm reading ofUtils.cpp again and will submit a PR with the includes in correct order. 
I've noticed   <mmsystem.h> was inserted before <direct.h>

**Comment by NickHardeman** [comment]
@dimitre ok, all I had to do to fix the Link Errors was add the `#pragma comment` mentioned above and it compiled. Not sure why that would be different between nightlies, I couldn't find the specified link to the lib in previous VS projects. I also tried adding the mmsystem.h and direct.h right in the ofDirectShowPlayer header without adding the .lib, but the linking errors still occurred.

**Comment by NickHardeman** [comment]
@dimitre see #7472

**Comment by ofTheo** [comment]
closed by #7472

**Comment by dimitre** [comment]
Great, thank you @NickHardeman
